src/XrdOuc/XrdOucCache.cc

```diff
@@ -44,12 +44,6 @@ int XrdOucCacheIO::pgRead(char                  *buff,
 {
    int bytes;
 
-// Make sure the offset is on a 4K boundary and the size is a multiple of
-// 4k as well (we use simple and for this).
-//
-   if ((offs & XrdSys::PageMask)
-   || (rdlen & XrdSys::PageMask)) return -EINVAL;
-
```

Comment:
How this check is being removed? While I agree the test for requiring all reads to be a multiple of 4K is likely to strict, for the ache all reads must be page aligned. So, at least that check should have remained.